src/test/scala/utils/KafkaTestContainer.scala

```diff
@@ -0,0 +1,63 @@
+package utils
+
+import cats.effect.syntax.all.*
+import cats.effect.{Ref, Resource, Sync}
+import cats.syntax.all.*
+import org.scalatest.Assertion
+import org.testcontainers.containers.KafkaContainer as Underlying
+import org.testcontainers.utility.DockerImageName
+import utils.KafkaContainer.KafkaConfig
+
+trait KafkaTestContainer[F[_]] {
+  def withRunningKafka(test: KafkaConfig => F[Assertion])(using Sync[F]): F[Assertion] =
+    KafkaContainer[F].use(running => test(running.config))
+}
+
+object KafkaContainer {
+  sealed trait Running[F[_]] {
+    def config: KafkaConfig
+    def stop: F[Unit]
+  }
+
+  def apply[F[_]](using F: Sync[F]): Resource[F, Running[F]] =
+    Resource.make {
+      for {
+        status           <- Ref.of(Status.Starting)
+        container        <- Either.catchNonFatal(Underlying(DockerImageName.parse("confluentinc/cp-kafka:7.4.0"))).liftTo[F]
```

Comment:
Is it worth having the docker image name/version in config or something? Seems a bit weird having it here directly

Reply:
I've also not used testcontainers before so not sure of conventions etc